lib/mail/network/delivery_methods/sendmail.rb

```diff
@@ -49,8 +49,11 @@ class DeliveryError < StandardError
     end
 
     def initialize(values)
+      if values[:arguments].is_a?(String)
+        deprecation_warn.call "Mail::Sendmail :arguments should be an Array of string args, not a string."
```

Comment:
This was my first take on a deprecation message and I think it reads more like an error message.
How about: `Initializing Mail::Sendmail with :arguments of type String is deprecated. Instead ensure :arguments is an array of string args, e.g. ["-i", "-t"].`